lib/activerecord-multi-tenant/model_extensions.rb

```diff
@@ -112,3 +112,11 @@ def inherited(subclass)
 if defined?(ActiveRecord::Base)
   ActiveRecord::Base.extend(MultiTenant::ModelExtensionsClassMethods)
 end
+
+class ActiveRecord::Associations::Association
+  alias skip_statement_cache_orig skip_statement_cache?
+  def skip_statement_cache?
+    return true if klass.respond_to?(:scoped_by_tenant?) && klass.scoped_by_tenant?
+    skip_statement_cache_orig
+  end
+end
```

Comment:
FYI `skip_statement_cache` takes a `scope` argument in version 5.2
https://github.com/rails/rails/blob/master/activerecord/lib/active_record/associations/association.rb#L276

You can work around that by using a splat, i.e.
```
  def skip_statement_cache?(*scope)
    return true if klass.respond_to?(:scoped_by_tenant?) && klass.scoped_by_tenant?
    skip_statement_cache_orig(*scope)
```